packages/esm-patient-labs-app/src/test-results/results-viewer/results-viewer.extension.tsx

```diff
@@ -65,6 +66,34 @@ const ResultsViewer: React.FC<ResultsViewerProps> = ({ patientUuid, basePath, lo
   const isExpanded = view === 'full';
   const trendlineView = testUuid && type === 'trendline';
   const responsiveSize = isTablet ? 'lg' : 'md';
+  const [showStickyHeader, setShowStickyHeader] = useState(false);
+  const headerRef = useRef(null);
+  const headerPositionRef = useRef(0);
+
+  useEffect(() => {
+    const setInitialPosition = () => {
+      if (headerRef.current) {
+        headerPositionRef.current = headerRef.current.getBoundingClientRect().top + window.scrollY;
+      }
+    };
+
+    setInitialPosition();
+
+    const handleScroll = () => {
+      if (headerRef.current) {
+        const scrollPosition = window.scrollY;
+        setShowStickyHeader(scrollPosition > headerPositionRef.current);
+      }
+    };
+
+    window.addEventListener('scroll', handleScroll);
+    window.addEventListener('resize', setInitialPosition);
+
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+      window.removeEventListener('resize', setInitialPosition);
+    };
+  }, []);
 
```

Comment:
I think it'd be preferable to achieve this using the [IntersectionObserver API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). It's more efficient as it's optimized by the browser and doesn't fire on every scroll. It also handles the visibility calculations internally, reducing the chance of errors. We could create an IntersectionObserver that triggers when the header is no longer visible in the viewport. We'd then update some state (something like `isHeaderVisible`) based on the intersection status. Here's how it could look:

```tsx
const [isHeaderVisible, setIsHeaderVisible] = useState(true);
  const headerRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    const observerCallback: IntersectionObserverCallback = (entries) => {
      const [entry] = entries;
      setIsHeaderVisible(entry.isIntersecting);
    };

    const observerOptions: IntersectionObserverInit = {
      threshold: 1,
      rootMargin: '-1px 0px 0px 0px',
    };

    const observer = new IntersectionObserver(observerCallback, observerOptions);

    const currentHeader = headerRef.current;
    if (currentHeader) {
      observer.observe(currentHeader);
    }

    return () => {
      if (currentHeader) {
        observer.unobserve(currentHeader);
      }
      observer.disconnect();
    };
  }, []);
```

And then we could replace `div.resultsHeader` with:

```tsx
<div ref={headerRef} className={styles.headerSentinel} />
<div className={classNames(styles.resultsHeader, { [styles.resultsHeaderScrolled]: !isHeaderVisible })}>
```

And have the following styling:

```tsx
.resultsHeader {
  padding: layout.$spacing-03 0;
  z-index: 10;
  background-color: $openmrs-background-grey;
  position: sticky;
  transition: border-bottom 0.2s ease-in-out;
}

.resultsHeaderScrolled {
  border-bottom: 1px solid colors.$gray-20;
  width: 100%;
  left: 0;
  right: 0;
}

.headerSentinel {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 1px;
  visibility: hidden;
}
```

Reply:
looking into this